iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -256,25 +349,40 @@ public static void checkBPerSection(Validable validable, double sectionB) {
 
     public static void checkMaximumSectionCount(Validable validable, int maximumSectionCount) {
         if (maximumSectionCount <= 0) {
-            throw new ValidationException(validable,
-                    "the maximum number of section (" + maximumSectionCount
-                            + ") should be greater than 0");
+            throw new ValidationException(validable, "the maximum number of section (" + maximumSectionCount
+                    + ") should be greater than 0");
         }
     }
 
-    public static void checkSections(Validable validable, int currentSectionCount, int maximumSectionCount) {
-        if (currentSectionCount < 0) {
-            throw new ValidationException(validable,
-                    "the current number of section (" + currentSectionCount
-                            + ") should be greater than or equal to 0");
-        }
+    public static ValidationLevel checkSections(Validable validable, Integer currentSectionCount, int maximumSectionCount) {
+        return checkSections(validable, currentSectionCount, maximumSectionCount, true);
+    }
+
+    public static ValidationLevel checkSections(Validable validable, Integer currentSectionCount, int maximumSectionCount, boolean throwException) {
+        return checkSections(validable, currentSectionCount, maximumSectionCount, throwException, Reporter.NO_OP);
+    }
+
+    public static ValidationLevel checkSections(Validable validable, Integer currentSectionCount, int maximumSectionCount, boolean throwException, Reporter reporter) {
```

Comment:
Indeed it is wierd that this method depends on maximumSectionCount.

Reply:
Not really as it checks if the current section count is feasible